src/RandArray.chpl

```diff
@@ -21,10 +21,10 @@ module RandArray {
         var seed = (seedStr:int) + here.id;
         fillRandom(a, seed);
       }
-      [ai in a] if (ai < 0) { ai = -ai; }
+      const a2 = a:uint;
       if (aMax > aMin) {
-        const modulus = aMax - aMin;
-        [x in a] x = ((x % modulus) + aMin):t;
+        const modulus = (aMax - aMin):uint;
```

Comment:
This `aMax - aMin` can still result in integer overflow, which is currently undefined behavior. e.g. `max(int) - (-1)` is undefined, not `min(int)`. 

Note that unsigned integer overflow is well defined and wraps. See https://github.com/chapel-lang/chapel/issues/6160 for more discussion on overflow.

We'll probably want to do something to convert to uint before doing the math. My mind goes to something like:

```chpl
const modules = if aMin > 0 then aMax:uint - aMin:uint else aMax:uint + abs(aMin):uint`
```

But that's not quite right since `abs(min(int))` is undefined as well (at least I think so, if it is defined the result is still `min(int)` which still causes problems.) Let me think on this for a bit and see what we do in our standard library for cases like this.

Reply:
This is the kind of question I've found Paul @cassella really excels at, if he's got any free cycles.  If I haven't mis-internalized his past guidance, my instinct would be that one could just write `(aMax:uint - aMin:uint) and get the right answer, assuming `aMax > aMin`—i.e., that any signed/unsigned wraparounds will simply do the right thing.  Paul, want to check my work?

Unrelated to this PR (and with the caveat that I don't know this code), seeing a modulus in the context of random number generation worries me slightly, though.  E.g., if I'm generating random 8-bit sequences and then using % to try and constrain them to be between 0..250, statistically speaking, I'm going to get twice as many entries in the 0..5 range than the 6..250 range due to wraparound, aren't I?  I think the more typical practice is to drop out-of-bound or "leftover" values on the floor and "re-roll".  I've lost track of the full interface of our random number module, but after a quick review of it the other week, I thought it had some capabilities along these lines.  Tagging @mppf here who is more expert in RNG than I am.

Reply:
remember we are trying to work like 
https://numpy.org/doc/stable/reference/random/generated/numpy.random.randint.html

Reply:
I agree with Brad's concern about `%`. (But it will not matter for all use cases). In the Random module we handle getting a bounded value with a re-roll that doesn't affect the RNG state. (see the getNext with min/mai here https://chapel-lang.org/docs/modules/standard/Random/PCGRandom.html#PCGRandom.PCGRandomStream.getNext ). We could potentially wire that up to `fillRandom` so you could give min/max arguments to `fillRandom`.

Reply:
that would be great but we still need a short term fix for this issue.

Reply:
Ethan and I could attempt to write the reroll and avoid using `%`. Or if a member of the chapel team writes the code we can add in that code as a workaround until that becomes the lowest supported chapel version

Reply:
I just pushed an update based on Brad's feedback.
@pierce314159 added an issue for tracking modulo bias and ensuring uniformity of the data (#1220).